cub/cub/util_macro.cuh

```diff
@@ -112,12 +112,12 @@ _CCCL_DIAG_SUPPRESS_CLANG("-Wattributes")
 #  if !defined(_CCCL_CUDA_COMPILER_NVHPC)
 _CCCL_DIAG_SUPPRESS_NVHPC(attribute_requires_external_linkage)
 #  endif // !_CCCL_CUDA_COMPILER_NVHPC
-#  if defined(_CCCL_COMPILER_ICC) || defined(_CCCL_COMPILER_ICC_LLVM)
```

Comment:
In the light of https://github.com/NVIDIA/cccl/issues/2314 can we just drop all of it here?

Reply:
But we continue to support ICC outside of cudax, AFAIU? Since this is CUB, we should keep it.